frontend/src/utils/date-utils.ts

```diff
@@ -99,23 +113,47 @@ export const dateStrToUpperCase = (dateStr: string): string => {
   return `${dateStr.slice(0, 1).toUpperCase()}${dateStr.slice(1)}`;
 };
 
-export const getDateFormat = (
-  date: number | string | undefined,
-  format: 'default' | 'snake',
+export const getFormattedDate = (
+  date: number | string | undefined | Date,
+  format:
+    | 'default'
+    | 'snake'
+    | typeof DEFAULT_DATE_FORMAT_SNAKE_CASE
+    | typeof DEFAULT_DATE_FORMAT
+    | 'YYYY-MM-DD HH:mm'
+    | 'DD_MM_YYYY'
+    | 'YYYY-MM-DDTHH:mm:ss',
 ) => {
   if (date === undefined) {
     return undefined;
   }
 
+  const jsDate = new Date(date);
+  const year = jsDate.getUTCFullYear();
+  const month = String(jsDate.getUTCMonth() + 1).padStart(2, '0');
+  const day = String(jsDate.getUTCDate()).padStart(2, '0');
+  const hours = String(jsDate.getUTCHours()).padStart(2, '0');
+  const minutes = String(jsDate.getUTCMinutes()).padStart(2, '0');
+  const seconds = String(jsDate.getUTCSeconds()).padStart(2, '0');
```

Comment:
These 3 lines are only needed in some cases, maybe put them behind a test to only compute them if needed, as they're probably quite expensive, relatively speaking.
Overall, this function is going to be called quite a lot, so I think it's worth speeding it up where possible.